controllers/machine_controller_phases.go

```diff
@@ -78,6 +78,13 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, m *clusterv1.
 	obj, err := external.Get(r.Client, ref, m.Namespace)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
+			if m.Status.InfrastructureReady && m.Status.GetTypedPhase() == clusterv1.MachinePhaseRunning {
+				// Infra object went missing after the machine was up and running
+				klog.Infof("Infrastructure object %v %q went missing. Updating Machine Status", ref.GroupVersionKind(), ref.Name)
+				machineStatusError := capierrors.MachineStatusError("Invalid Configuration")
```

Comment:
@vincepri do we have any guidance on the content and format for error reasons?

Reply:
We should avoid creating new ErrorReason(s) dynamically. We can add new ones to the constants defined in the package. Let's focus on having a good error message in `ErrorMessage` though